File: examples/access/constraints/off_nadir_side_looking.py
# /// script
# dependencies = ["brahe"]
# ///
"""
Create an off-nadir constraint for side-looking radar requiring specific viewing geometry.
"""

import brahe as bh

# Side-looking radar requiring specific geometry
constraint = bh.OffNadirConstraint(min_off_nadir_deg=20.0, max_off_nadir_deg=45.0)

print(f"Created: {constraint}")
# Created: OffNadirConstraint(20.0° - 45.0°)

File: examples/access/constraints/off_nadir_side_looking.rs
//! Create an off-nadir constraint for side-looking radar requiring specific viewing geometry.

#[allow(unused_imports)]
use brahe as bh;

fn main() {
    // Side-looking radar
    let constraint = bh::OffNadirConstraint::new(
        Some(20.0),
        Some(45.0)
    ).unwrap();

    println!("Created: {}", constraint);
    // Created: OffNadirCOffNadirConstraint(20.0° - 45.0°)
}
